File: programacao/c/respostas/operacoes/34.c
#include <stdio.h>

int main() {
    float limite_inf_orig, limite_sup_orig, valor_lido;
    float limite_inf_dest, limite_sup_dest;
    float valor_convertido;

    printf("Digite o limite inferior da escala original: ");
    scanf("%f", &limite_inf_orig);

    printf("Digite o limite superior da escala original: ");
    scanf("%f", &limite_sup_orig);

    printf("Digite o valor lido na escala original: ");
    scanf("%f", &valor_lido);

    printf("Digite o limite inferior da escala de destino: ");
    scanf("%f", &limite_inf_dest);

    printf("Digite o limite superior da escala de destino: ");
    scanf("%f", &limite_sup_dest);

    valor_convertido = limite_inf_dest +
                      (valor_lido - limite_inf_orig) *
                      (limite_sup_dest - limite_inf_dest) /
                      (limite_sup_orig - limite_inf_orig);

    printf("O valor convertido e: %.2f\n", valor_convertido);

    return 0;
}

File: programacao/python/respostas/operacoes/34.py
limite_inf_orig = float(input("Limite inferior da escala original: "))
limite_sup_orig = float(input("Limite superior da escala original: "))
valor_lido = float(input("Valor lido na escala original: "))
limite_inf_dest = float(input("Limite inferior da escala de destino: "))
limite_sup_dest = float(input("Limite superior da escala de destino: "))

valor_convertido = limite_inf_dest + (valor_lido - limite_inf_orig) * (
    limite_sup_dest - limite_inf_dest
) / (limite_sup_orig - limite_inf_orig)

print(f"O valor convertido e: {valor_convertido:.2f}")
